app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/provision/cloning.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'json'
+
+# Provide all configurable aspects of the actual provision operation.
+module ManageIQ::Providers::IbmCloud::VPC::CloudManager::Provision::Cloning
+  # Called during provision. Prints message to log file.
+  # @param clone_options [Hash] Options used for clone operation.
+  # @return [Boolean] The result of the log message.
+  def log_clone_options(clone_options)
+    json_clone_options = JSON.dump(clone_options)
+    _log.info("IBM SERVER PROVISIONING OPTIONS: #{json_clone_options}")
+  rescue => e
+    logger(__method__).log_backtrace(e, :context_msg => 'Error while logging clone options for VPC provision.')
+  end
+
+  # Send the final product to the Cloud provider.
+  # @param clone_options [Hash] Payload to send to provider.
+  # @raise [MiqException::MiqProvisionError] An error was returned by the SDK.
+  # @return [String] The ID of the new instance.
+  def start_clone(clone_options)
+    logger(__method__).debug("Json for VPC provision task. #{JSON.dump(clone_options)}")
```

Comment:
The `Vmdb::LogProxy` class dynamically determines the calling method with `#caller_locations` so if you use that you shouldn't have to pass in `__method__`

Reply:
I'm not seeing duplication in my logs, and so I don't think that is true.

```
[----] I, [2021-04-01T15:19:41.884962 #51274:3fdf4aadae08]  INFO -- : ManageIQ::Providers::IbmCloud::VPC::CloudManager::ProvisionWorkflow.parse_new_volumes_fields Values are {:initial_pass=>true, :src_vm_id=>[36, "ibm-centos-7-0-minimal-amd64-1"], :miq_request_dialog_name=>"miq_provision_vpc_dialogs", :current_tab_key=>:volumes, :ip_addr=>nil, :user_script_text=>nil, :provision_type=>[1, "bx2-2x8"], :vm_name=>"avdd", :placement_auto=>[false, 0], :placement_availability_zone=>[1, "us-east-1"], :cloud_network=>["r014-0fa2acc6-2a41-4f2b-9c89-bcea07cdcbc3", "b-vpc-washington-dc"], :cloud_subnet=>["0757-ef523a2f-5356-42ff-8a78-9325509465b9", "b-subneet-washington-dc-1"], :resource_group=>nil, :storage_type=>["general-purpose", "general-purpose"], :guest_access_key_pair=>["r014-14c6c21a-a33d-4323-863b-439561483102", "b-goo-washington-dc-key"], :cloud_volumes=>[nil, nil], :volume_name=>nil, :volume_resource_group=>"default", :volume_size=>nil, :volume_profile=>"general-purpose", :vm_tags=>[], :src_vm_nics=>nil, :src_vm_lans=>[], :src_ems_id=>[2, "Test"], :schedule_time=>nil, :volume_name_1=>"abc", :volume_resource_group_1=>"abc", :volume_profile_1=>"abc", :volume_size_1=>"20"}
```

Reply:
```
p = ManageIQ::Providers::IbmCloud::VPC::CloudManager::Provision.new
>> p.logger("foobar").info("hello")
```

evm.log
```
INFO -- : MIQ(ManageIQ::Providers::IbmCloud::VPC::CloudManager::LoggingMixin::LoggingWrapper#info) ManageIQ::Providers::IbmCloud::VPC::CloudManager::Provision.foobar hello
```

Reply:
True, evm log does print out what it thinks is the calling method. Whereas LoggingWrapper has the expected method.

Reply:
Now that you aren't logging to `_log` this isn't an issue anymore

Reply:
The intended target for the default message is still _log. For debugging the target is $ibm_cloud_log to provide an order of events in one place. Essentially, during normal operation _log logs the operation which is good because all tthe other messages are there. During debugging $ibm_cloud_log has a record which is good because all the exceptions are there.

Reply:
@jaredm-ibm I'm not sure what you mean, I'm stating that now that you are no longer passing a formatted message that includes the calling class+method to `_log` that this no longer will double-prefix messages which is what my comment is about.

If I'm wrong about that and LoggingWrapper actually still is logging to `_log` then please correct this in a follow-up